### PR TITLE
#163426525 View and edit user profile

### PR DESCRIPTION
**Pull request by primuse**
#### What does this PR do?
It creates API endpoints to
- Verify user account
- View user profile
- Edit user profile
- Edit user picture
#### Description of Task to be completed?
Have the following endpoints working:
- Verify user account: `PATCH /api/v1/user/verify`
- View user profile: `GET /profiles/:username`
- Edit user profile: `PATCH /profiles/:username/update`
- Edit user picture: `PATCH /profiles/image`
#### How should this be manually tested?
Clone this repo, cd into it and run npm install.
 **With test suite:**
     - Run `npm test`

 **Using postman:**
     - Run `npm run dev`,
     - With postman, send requests to `localhost:3000/<api url>` with the relevant HTTP methods and request payloads.
#### What are the relevant pivotal tracker stories?
[#163426525](https://www.pivotaltracker.com/story/show/163426525)

### CODE REVIEW

**Review by divinediscipline — DISMISSED** [review]
Looks good. Some little changes are required in the source-json file, looks like the username and bio appears twice.

**Review by omenkish — DISMISSED** [review]

**Review by biolabalo — DISMISSED** [review]

**Review by EfeAgare — DISMISSED** [review]

**Review by divinediscipline — DISMISSED** [review]
Looks good to me

File: server/db/models/users.js [hunk 1]
```diff
@@ -56,9 +56,11 @@ export default (sequelize, DataTypes) => {
           user.dataValues.password = await bcrypt.hash(user.dataValues.password, saltRounds);
           return user;
         },
-        beforeUpdate: async (user) => {
+        beforeBulkUpdate: async (user) => {
```
Comment (p-funky) [inline server/db/models/users.js:59]: Why was it changed to `beforeBulkUpdate`?

File: server/helpers/jwtHelper.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   * @static
+   * @param {String} token to be verified
+   * @returns {Object} a user payload or an error
+   */
+  static verifyToken(token) {
+    if (!token) return false;
+    try {
+      const decodedToken = jwt.verify(token, process.env.JWT_SECRET_KEY);
+      return decodedToken;
+    } catch (e) {
```
Comment (p-funky) [inline server/helpers/jwtHelper.js]: Yeah we all know what `e` is but let's just be more descriptive and call it `error`.

File: server/services/email.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ sgMail.setApiKey(process.env.SENDGRID_API_KEY);
 const sendEmail = (receiverEmail, subject, html) => {
   sgMail.send({
     to: receiverEmail,
-    from: 'AuthorsHaven@Andela.com',
+    from: 'AuthorsHaven@andela.com',
```
Comment (p-funky) [inline server/services/email.js:19]: Nice of you to catch this.

File: server/tests/app/auth.test.js [hunk 4]
```diff
@@ -98,7 +98,7 @@ describe('user registration', () => {
 
     .then((res) => {
       expect(res.status).to.equal(201);
-      expect(res.body.message).to.equal('successfully registered to authors haven');
+      expect(res.body.message).to.equal('Successfully registered to Authors haven. Kindly check your Email to verify your account');
```
Comment (p-funky) [inline server/tests/app/auth.test.js]: More descriptive message here, cool. But change Email to email though.

File: server/db/models/users.js (hunk 1 above)
Comment (primuse) [reply server/db/models/users.js:59]: The hook wasn't being hit when updating a user. So I researched on it and found this: `beforeUpdate` hook called when using `instance.save()` or `instance.update()`, when using `model.update()`, `beforeBulkUpdate` will be called.

Comment (p-funky) [reply server/db/models/users.js:59]: Alright, gotcha. Must have changed somewhere in between versions I suspect. I guess this was tested yeah?

Comment (primuse) [reply server/db/models/users.js:59]: yep, it was

**Review by p-funky — APPROVED** [review]

**Review by EfeAgare — APPROVED** [review]

**Review by omenkish — APPROVED** [review]

**Pull request merged**